domain/test/usecases/authentication/create_permanent_token_interactor_test.dart

```diff
@@ -0,0 +1,114 @@
+      createPermanentTokenInteractor = CreatePermanentTokenInteractor(authenticationRepository, tokenRepository, credentialRepository);
+    });
+
+    test('createPermanentTokenInteractorShouldReturnSuccessWithCorrectData', () async {
+      when(authenticationRepository.createPermanentToken(linShareBaseUrl, userName1, password1))
+          .thenAnswer((_) async => permanentToken);
+      final result = await createPermanentTokenInteractor.execute(linShareBaseUrl, userName1, password1);
+      expect(result, Right<Failure, Success>(AuthenticationViewState(permanentToken)));
+    });
+
+    test('createPermanentTokenInteractorShouldFailureWithWrongUrl', () async {
+      when(authenticationRepository.createPermanentToken(wrongUrl, userName1, password1))
+          .thenThrow(ServerNotFound());
+      final result = await createPermanentTokenInteractor.execute(wrongUrl, userName1, password1);
+      expect(result, Left<Failure, Success>(AuthenticationFailure(ServerNotFound())));
+    });
+
+    test('createPermanentTokenInteractorShouldFailureWithWrongUserName', () async {
+      when(authenticationRepository.createPermanentToken(linShareBaseUrl, userName2, password1))
+          .thenThrow(BadCredentials());
+      final result = await createPermanentTokenInteractor.execute(linShareBaseUrl, userName2, password1);
+      expect(result, Left<Failure, Success>(AuthenticationFailure(BadCredentials())));
+    });
+
+    test('createPermanentTokenInteractorShouldFailureWithWrongPassword', () async {
+      when(authenticationRepository.createPermanentToken(linShareBaseUrl, userName1, password2))
+          .thenThrow(BadCredentials());
+      final result = await createPermanentTokenInteractor.execute(linShareBaseUrl, userName1, password2);
+      expect(result, Left<Failure, Success>(AuthenticationFailure(BadCredentials())));
+    });
+
+    test('createPermanentTokenInteractorShouldFailureWithConnectionError', () async {
+      when(authenticationRepository.createPermanentToken(linShareBaseUrl, userName1, password1))
+          .thenThrow(ConnectError());
+      final result = await createPermanentTokenInteractor.execute(linShareBaseUrl, userName1, password1);
+      expect(result, Left<Failure, Success>(AuthenticationFailure(ConnectError())));
+    });
+
+    test('createPermanentTokenInteractorShouldFailureWithUnknownError', () async {
+      when(authenticationRepository.createPermanentToken(linShareBaseUrl, userName1, password1))
+          .thenThrow(UnknownError('unknown error'));
+      final result = await createPermanentTokenInteractor.execute(linShareBaseUrl, userName1, password1);
+      expect(result, Left<Failure, Success>(AuthenticationFailure(UnknownError('unknown error'))));
+    });
+
+    test('createPermanentTokenInteractorShouldFailureWithInternalServerError', () async {
+      when(authenticationRepository.createPermanentToken(linShareBaseUrl, userName1, password1))
+          .thenThrow(InternalServerError());
```

Comment:
Really?, as I understand, with this kind of error, we will throw `UnknownError`? Am I right?

Reply:
Oh I think this test is correct but I missed logic code for this case in `AuthenticationDataSource`. Let me add that.